common/src/state/ui.rs

```diff
@@ -323,13 +323,27 @@ impl ToastNotification {
     }
 }
 
-#[derive(PartialEq, Clone, Debug, Default, Deserialize, Serialize)]
+#[derive(Eq, Clone, Debug, Default, Deserialize, Serialize)]
 pub struct Theme {
     pub filename: String,
     pub name: String,
     pub styles: String,
 }
 
+impl Theme {
+    pub fn remove_duplicates_and_sort_by_name(themes: &mut Vec<Theme>) -> Vec<Theme> {
```

Comment:
Would be best to let this function own `themes` than to borrow it

```suggestion
    pub fn remove_duplicates_and_sort_by_name(mut themes: Vec<Theme>) -> Vec<Theme> {
```

Reply:
Although maybe we should move the logic here into `get_available_themes` since this is the only purpose and may not be used in other parts of the UI?